### ISSUE TITLE
Rule proposal: disallow onFocusIn and onFocusOut on DOM element tags

### ISSUE THREAD

**Issue by lencioni** [yannickcr/eslint-plugin-react#714]
React will issue a warning in dev if you use `onFocusIn` or `onFocusOut`.

https://github.com/facebook/react/blob/85dcbf83/src/renderers/dom/shared/ReactDOMComponent.js#L196-L200

This should be easy enough to enforce for most cases with a linter rule.

Bad:

```jsx
<div onFocusIn={() => {}} />
```
```jsx
<div onFocusOut={() => {}} />
```

Good:

```jsx
<div onFocus={() => {}} />
```

```jsx
<div onBlur={() => {}} />
```

**Comment by lencioni** [comment]
Name: `dom-elements-no-on-focus-in-or-out`?
